src/lib/db.js

```diff
@@ -63,33 +71,32 @@ function wipeDb (db, path, cb) {
  * closing and wiping the db beforehand and re-upping it afterwards.
  */
 function importBulkOsm (bbox, cb) {
-  const queryUrl = url.resolve(osmapi, `map?bbox=${bbox}`)
-  const importFn = () => {
-    importer(osmOrgDbPath, request(queryUrl), function (err) {
-      osmOrgDb = osmOrgDb || osmdb(osmOrgDbPath)
-      if (err) {
-        console.warn(err)
-        cb(err)
-      } else {
-        // Deduplicate nodes that appear in the new OSM.org data *and* the observationsDb.
-        deduplicatePlaceholderNodes(observationsDb, osmOrgDb, observationsIndex, function (err) {
-          cb(err, `Finished importing ${bbox}`)
+  wipeDb(osmOrgDb.db, osmOrgDbPath, (err) => {
+    if (err) {
+      done(err)
+    } else {
+      getOsmStream(bbox, function (err, stream) {
+        if (err) return done(err)
+        importer(osmOrgDbPath, stream, function (err) {
+          if (err) {
+            done(err)
+          } else {
+            // up the DB before deduping nodes
+            osmOrgDb = osmOrgDb || osmdb(osmOrgDbPath)
+            // Deduplicate nodes that appear in the new OSM.org data *and* the observationsDb.
+            deduplicatePlaceholderNodes(observationsDb, osmOrgDb, observationsIndex, function (err) {
+              done(err, `Finished importing ${bbox}`)
+            })
+          }
         })
-      }
-    })
-  }
+      })
+    }
+  })
 
-  if (!osmOrgDb) importFn()
-  else {
-    wipeDb(osmOrgDb.db, osmOrgDbPath, (err) => {
-      if (err) {
-        console.warn(err)
-        cb(err)
-      } else {
-        osmOrgDb = null
-        importFn()
-      }
-    })
+  function done (err, message) {
+    osmOrgDb = osmOrgDb || osmdb(osmOrgDbPath)
```

Comment:
Yes! Nice catch. I had just written a comment about this, and then saw you handled making sure `osmOrgDb` always points to its DB. :)